lib/backend/firestore/firestorebk.go

```diff
@@ -436,23 +471,31 @@ func (b *Backend) getRangeDocs(ctx context.Context, startKey, endKey backend.Key
 		limit = backend.DefaultRangeLimit
 	}
 	docs, err := b.svc.Collection(b.CollectionName).
-		Where(keyDocProperty, ">=", startKey).
-		Where(keyDocProperty, "<=", endKey).
+		Where(keyDocProperty, ">=", []byte(startKey.String())).
+		Where(keyDocProperty, "<=", []byte(endKey.String())).
 		Limit(limit).
 		Documents(ctx).GetAll()
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 	legacyDocs, err := b.svc.Collection(b.CollectionName).
-		Where(keyDocProperty, ">=", string(startKey)).
-		Where(keyDocProperty, "<=", string(endKey)).
+		Where(keyDocProperty, ">=", startKey.String()).
+		Where(keyDocProperty, "<=", endKey.String()).
+		Limit(limit).
+		Documents(ctx).GetAll()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	brokenDocs, err := b.svc.Collection(b.CollectionName).
+		Where(keyDocProperty, ">=", startKey).
+		Where(keyDocProperty, "<=", endKey).
```

Comment:
are you sure this won't get the same set of docs as the []bytes range query above? backend.Key is also just a byte slice, and I don't know how this code works